packages/admin/admin/src/form/FinalFormInput.tsx

```diff
@@ -1,28 +1,50 @@
-import { InputBase, InputBaseProps } from "@mui/material";
+import { Button, InputBase, InputBaseProps } from "@mui/material";
 import * as React from "react";
 import { FieldRenderProps } from "react-final-form";
+import { FormattedMessage } from "react-intl";
 
 import { ClearInputAdornment } from "../common/ClearInputAdornment";
+import { useContentTranslationServiceProvider } from "../translator/config/useContentTranslationServiceProvider";
 
 export type FinalFormInputProps = InputBaseProps &
     FieldRenderProps<string, HTMLInputElement | HTMLTextAreaElement> & {
         clearable?: boolean;
+        disableTranslation?: boolean;
     };
 
-export function FinalFormInput({ meta, input, innerRef, endAdornment, clearable, ...props }: FinalFormInputProps): React.ReactElement {
+export function FinalFormInput({
+    meta,
+    input,
+    innerRef,
+    endAdornment,
+    clearable,
+    disableTranslation,
+    ...props
+}: FinalFormInputProps): React.ReactElement {
+    const { enabled, translate } = useContentTranslationServiceProvider();
+
     return (
         <InputBase
             {...input}
             {...props}
             endAdornment={
-                clearable ? (
-                    <>
+                <>
+                    {clearable && (
                         <ClearInputAdornment position="end" hasClearableContent={Boolean(input.value)} onClick={() => input.onChange("")} />
-                        {endAdornment}
-                    </>
-                ) : (
-                    endAdornment
-                )
+                    )}
+                    {enabled && !disableTranslation && (
+                        <Button
+                            onClick={async () => {
+                                if (translate) {
+                                    input.onChange(await translate(input.value));
+                                }
```

Comment:
Out of scope: We'll need better error handling here

Reply:
Ah, the if (translate) is unnecessary since translate now has to always be defined.